### PR TITLE
GG-32007 Fixed erroneous synchronization for accessing transaction configuration

### PR DESCRIPTION
**Pull request by sk0x50**

### CODE REVIEW

File: modules/opencensus/src/test/java/org/apache/ignite/internal/processors/monitoring/opencensus/OpenCensusDiscoveryTracingTest.java [hunk 1]
```diff
@@ -116,7 +116,8 @@ public void testClientNodeCacheCreateTracing() throws Exception {
                 catch (Exception e) {
                     fail("Unable to create cache.");
                 }
-            }
+            },
+            GRID_CNT
```
Comment (sanpwc) [inline modules/opencensus/src/test/java/org/apache/ignite/internal/processors/monitoring/opencensus/OpenCensusDiscoveryTracingTest.java:120]: As far as I understand it was made in order to overcome the problem of configuration unavailability on client node, cause of "async" nature of metastore data propagation to the client nodes.
Despite the fact that it's valid to use same node for both setting configuration and trigger operation to be traced, I'd rather use wait for condition approach within some utility method. What's more important we should ensure that any of given approaches is used everywhere but not only within given test. For example OpenCensusTxTracingTest also should be modified.
In other words I suggest to add some sort of `ensureTracingConfigurationWasSet(expectedConfig, cluster) ` that will be triggered after setting configuration and starting client nodes.

File: modules/opencensus/src/test/java/org/apache/ignite/internal/processors/monitoring/opencensus/OpenCensusDiscoveryTracingTest.java [hunk 2]
```diff
@@ -286,8 +290,25 @@ public void testClientNodeStopTracing() throws Exception {
...
+    ) throws Exception {
+        performClientBasedOperation(preparationOperations, operationToCheck, 0);
+    }
+
+    /**
+     * Perform preparation operations and operation to trigger span creation.
+     *
+     * @param preparationOperations Operations to run without tracing.
+     * @param operationToCheck Operations to run with tracing.
+     * @param nodeIdx Node id that that is used for changing tracing configuration.
```
Comment (sanpwc) [inline modules/opencensus/src/test/java/org/apache/ignite/internal/processors/monitoring/opencensus/OpenCensusDiscoveryTracingTest.java:303]: If you'll continue with same-node-solution please add more meaningful comment that will describe why it's needed.

Comment (sk0x50) [reply modules/opencensus/src/test/java/org/apache/ignite/internal/processors/monitoring/opencensus/OpenCensusDiscoveryTracingTest.java:303]: Ok, will do.

File: modules/opencensus/src/test/java/org/apache/ignite/internal/processors/monitoring/opencensus/OpenCensusDiscoveryTracingTest.java (hunk 1 above)
Comment (sk0x50) [reply modules/opencensus/src/test/java/org/apache/ignite/internal/processors/monitoring/opencensus/OpenCensusDiscoveryTracingTest.java:120]: Ok, it makes sense to me. I will try to modify tests in that way.
